server/interface.go

```diff
@@ -12,6 +12,7 @@ type Client interface {
 
 	FindProposals(providerID string) (proposals []dto_discovery.ServiceProposal, err error)
 	RegisterProposal(proposal dto_discovery.ServiceProposal, signer identity.Signer) (err error)
+	UnregisterProposal(providerId string, signer identity.Signer) (err error)
```

Comment:
Be consistent with another method in interface `PingProposal(proposal)`:
- pass full `proposal` structure
- real adapted takes that he needs for unregistration DTO